app/routers/search_endpoint.py

```diff
@@ -0,0 +1,72 @@
+from fastapi import APIRouter, HTTPException, Query
+from app.services.openai import OpenAIService
+from ..shared_resources import document_data_store
+from numpy import dot
+from numpy.linalg import norm
+
+router = APIRouter()
+_THRESHOLD = 0.80
+
+
+@router.post("/search/{query}")
+async def semantic_search(
+    query: str,
+    ai_provider: str = Query("openai", description="AI provider for semantic search"),
+):
```

Comment:
As per our current implementation of shared resources, the user has to pass the pageid in the endpoint so the backend can find the corresponding text from `page_data_store`